georchestra/georchestra
# Docker remove georchestra base

The goal of this PR is to remove an extra layer with the `georchestra/base`. It also adds a definition of the volumes in order to allow data persistence.

The use of a georchestra/base image actually introduces some complexity to customize images afterwards, and forces the image to be generated with the complete datadir copied into /etc/georchestra which is unecessary.

The PR addresses also the need for people willing to use docker and already have a git repository to use it instead of the default one, by issuing the following command:

```
mvn clean package docker:build -DdockerDatadirScmUrl=scm:git:git@example.com:myprivaterepo/datadir.git -DdockerDatadirScmVersion=mycustombranch
```

Note: when merging this into master in the next cascade merge, it will require to modify the poms to change the `dockerDatadirScmVersion` to `docker-official` instead of `docker-15.12`.

Note2: once the PR onto geonetwork related to this one (https://github.com/georchestra/geonetwork/pull/36), it will require to bump the geonetwork submodule version.

Note3: the previous behaviour (mvn clean package docker:build ...) should stay iso-functional if this PR is merged, the main added value is that the parameters `-Dvariable=...` have been added for convenience.

Tests status (runtime on locally generated containers):
- [x] mapfishapp
- [x] extractorapp
- [x] security-proxy
- [x] geowebcache
- [x] ldapadmin
- [x] geoserver
- [x] geonetwork 2
- [x] geonetwork 3.0.x
- [x] catalogapp
- [x] downloadform

Test protocol:

- `mvn clean package docker:build -Pdocker` on each module
- `docker run georchestra/<module>`
- check the docker output
- connect onto the container and check that the `/etc/georchestra` exists as expected, and contains only the subdirectory for the given module.
